cosmos/x/evm/plugins/block/genesis.go

```diff
@@ -26,21 +26,31 @@ import (
 	"pkg.berachain.dev/polaris/eth/core"
 )
 
-// InitGenesis performs genesis initialization for the evm module. It returns
-// no validator updates.
-func (p *plugin) InitGenesis(ctx sdk.Context, _ *core.Genesis) {
-	// TODO: IMPLEMENT
+// InitGenesis performs genesis initialization for the block plugin in the
+// evm module.
+func (p *plugin) InitGenesis(ctx sdk.Context, ethGen *core.Genesis) {
 	p.Prepare(ctx)
-	// p.StoreHeader(ethGen.ToBlock().Header())
+	if err := p.StoreHeader(ethGen.ToBlock().Header()); err != nil {
```

Comment:
id just store directly on the blockchain imo to avoid the context weirrdness issue,